mesa/experimental/jupyter_viz.py

```diff
@@ -51,6 +51,23 @@ def make_model():
         set_current_step(0)
         return model
 
+    def calculate_space_size():
+        if model_parameters["N"] > 50:
+            fixed_size = int(model_parameters["N"] / 5)
+            model_parameters.update({"width": fixed_size, "height": fixed_size})
+            plt.rcParams["figure.figsize"] = (fixed_size, fixed_size)
+
+    def calculate_agent_size():
+        aspect_ratio_high = (
+            0.5  # maintain this aspect ration b/w agent size and space size
+        )
+        aspect_ratio_low = 0.1  #
+        print()
+        size_of_agent = aspect_ratio_high * (
+            model_parameters["width"] * model_parameters["height"]
```

Comment:
(Must be GitHub hiccup. I remembered already commenting about this) the size is length^1, but the width * height is length^2, so it scales quadratically instead. It makes more sense to scale linearly over `min(width, height)`.

Reply:
yes, the idea is to have a constant ratio b/w size of agents (for e.g. area of a circle) and size of the figure (area of a square). Hence the scaling is still linear i.e. the size of agent = aspect_ratio * size of figure.

Reply:
I just realized that the `s` param in Matplotlib scatter is length^2: https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.scatter.html. It's `markersize**2` without a multiplication by pi. We should document that size refers to Matplotlib size, and it is an area.